ambry-store/src/test/java/com.github.ambry.store/CompactionManagerTest.java

```diff
@@ -17,217 +17,136 @@
 import com.github.ambry.config.StoreConfig;
 import com.github.ambry.config.VerifiableProperties;
 import com.github.ambry.utils.MockTime;
-import com.github.ambry.utils.TestUtils;
 import com.github.ambry.utils.Time;
-import com.github.ambry.utils.UtilsTest;
-import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Properties;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
 import static org.junit.Assert.*;
 
 
 /**
- * Unit tests {@link CompactionManager}.
+ * Unit tests {@link CompactionManager}
  */
+@RunWith(Parameterized.class)
 public class CompactionManagerTest {
 
-  private static long CAPACITY_IN_BYTES = 10 * 1024 * 1024;
-  private static long DEFAULT_USED_CAPACITY_IN_BYTES = CAPACITY_IN_BYTES * 6 / 10;
-  // the properties that will used to generate a StoreConfig. Clear before use if required.
-  private final Properties properties = new Properties();
-  private final Time time = new MockTime();
-  private StoreConfig config;
+  private Properties properties = new Properties();
+  private final StoreConfig config;
+  private final Time time;
   private MockBlobStore blobStore;
+  private final CompactionPolicyTest.CompactionPolicyType type;
   private CompactionManager compactionManager;
-  private long messageRetentionTimeInMs;
 
   /**
-   * Instantiates {@link CompactionManagerTest} with the required cast
-   * @throws InterruptedException
+   * Running for both CompactAll and Default Compaction Policy
+   * @return an array with both {@code CompactionPolicyType.DEFAULT} and {@code CompactionPolicyType.COMPACT_ALL}.
```

Comment:
why do we need this to run with two policies? This looks like an integration test to me. If anything, we should mock the policy (or use compact all) for the tests here.